tests/MongoDB.Driver.Tests/Linq/Linq2ImplementationTestsOnLinq3/Translators/AggregateGroupTranslatorTests.cs

```diff
@@ -38,7 +39,10 @@ public void Should_translate_using_non_anonymous_type_with_default_constructor()
         {
             var result = Group(x => x.A, g => new RootView { Property = g.Key, Field = g.First().B });
 
-            result.Projection.Should().Be("{ $project : { Property : '$_id', Field : { $let : { vars : { this : { $arrayElemAt : [ '$_elements', 0 ] } }, in : '$$this.B' } }, _id : 0 } }");
+            AssertStages(
+                result.Stages,
+                "{ $group : { _id : '$A', __agg0 : { $first : '$$ROOT' } } }",
+                "{ $project : { Property : '$_id', Field : '$__agg0.B', _id : 0 } }");
```

Comment:
I've changed the tests to check BOTH the $group and $project stages because they have to work together.

Please check all the calls to `AssertStages` carefully because these are not integration tests, so if we're generating invalid MQL we're not detecting that.